cartography/intel/gcp/pubsub.py

```diff
@@ -0,0 +1,232 @@
+                    subscription['region'] = 'global'
+                    subscription['id'] = subscription['name']
+                    subscription['subscription_name'] = subscription.get('name').split('/')[-1]
+                    subscriptions.append(subscription)
+            req = pubsub.projects().subscriptions().list_next(previous_request=req, previous_response=res)
+
+        if common_job_parameters.get('pagination', {}).get('pubsub', None):
+            pageNo = common_job_parameters.get("pagination", {}).get("pubsub", None)["pageNo"]
+            pageSize = common_job_parameters.get("pagination", {}).get("pubsub", None)["pageSize"]
+            totalPages = len(subscriptions) / pageSize
+            if int(totalPages) != totalPages:
+                totalPages = totalPages + 1
+            totalPages = int(totalPages)
+            if pageNo < totalPages or pageNo == totalPages:
+                logger.info(f'pages process for pubsub subscriptions {pageNo}/{totalPages} pageSize is {pageSize}')
+            page_start = (common_job_parameters.get('pagination', {}).get('pubsub', None)[
+                          'pageNo'] - 1) * common_job_parameters.get('pagination', {}).get('pubsub', None)['pageSize']
+            page_end = page_start + common_job_parameters.get('pagination', {}).get('pubsub', None)['pageSize']
+            if page_end > len(subscriptions) or page_end == len(subscriptions):
+                subscriptions = subscriptions[page_start:]
+            else:
+                has_next_page = True
+                subscriptions = subscriptions[page_start:page_end]
+                common_job_parameters['pagination']['pubsub']['hasNextPage'] = has_next_page
+
+        return subscriptions
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve pubsub subscriptions on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+
+@timeit
+def load_pubsub_subscriptions(session: neo4j.Session, data_list: List[Dict], project_id: str, update_tag: int) -> None:
+    session.write_transaction(load_pubsub_subscriptions_tx, data_list, project_id, update_tag)
+
+
+@timeit
+def load_pubsub_subscriptions_tx(
+    tx: neo4j.Transaction, data: List[Dict],
+    project_id: str, gcp_update_tag: int,
+) -> None:
+
+    query = """
+    UNWIND {Records} as record
+    MERGE (subscription:GCPPubsubSubscription{id:record.id})
+    ON CREATE SET
+        subscription.firstseen = timestamp()
+    SET
+        subscription.lastupdated = {gcp_update_tag},
+        subscription.region = record.region,
+        subscription.name = record.subscription_name,
+        subscription.topic = record.topic,
```

Comment:
using this topic field, we should build a relationship between topic & subscription. relationship can be HAS